integration/src/test/java/org/mule/test/core/context/notification/PipelineMessageNotificationTestCase.java

```diff
@@ -118,20 +118,19 @@ protected String getConfigFile() {
   public PipelineMessageNotificationTestCase(String caseName, String flowName, Class<? extends Throwable> expectedExceptionClass,
                                              RestrictedNode spec, Consumer<PipelineMessageNotificationTestCase> assertions) {
     this.flowName = flowName;
-    this.expectedExceptionClz = expectedExceptionClass;
+    if (expectedExceptionClass != null) {
+      this.expectedException.expectCause(isA(expectedExceptionClass));
```

Comment:
there is a matcher for error types in the test utils, so that you can match the specific error type raised instead fo the generic exception class.